src/Microsoft.AspNetCore.OData/Edm/EdmPrimitiveHelper.cs

```diff
@@ -105,8 +105,12 @@ public static object ConvertPrimitiveValue(object value, Type type, TimeZoneInfo
                 {
                     DateTimeOffset dateTimeOffsetValue = (DateTimeOffset)value;
                     TimeZoneInfo timeZone = timeZoneInfo ?? TimeZoneInfo.Local;
+
                     dateTimeOffsetValue = TimeZoneInfo.ConvertTime(dateTimeOffsetValue, timeZone);
-                    return dateTimeOffsetValue.DateTime;
+
+                    var dateTimeKind = GetTargetDateTimeKind(timeZone);
```

Comment:
We prefer `explicit type` instead of `var`
```suggestion
                    DateTimeKind dateTimeKind = GetTargetDateTimeKind(timeZone);
```